File: EntropieReductor/app/src/main/java/de/frank/entropyreducer/data/remote/tts/GoogleTtsVoices.kt
package de.frank.entropyreducer.data.remote.tts

/**
 * Liste der deutschen Chirp 3 HD-Stimmen von Google Cloud TTS.
 *
 * Quelle: cloud.google.com/text-to-speech/docs/chirp3-hd
 * Kosten: 1 Mio Zeichen/Monat gratis, danach $30/Mio.
 *
 * Default `de-DE-Chirp3-HD-Kore` (weiblich, warm) — uebernommen aus
 * BestJournalFrank wo dieselben Stimmen produktiv im Einsatz sind.
 */
data class GoogleTtsVoice(
    val name: String,
    val displayName: String,
    val gender: VoiceGender,
)

enum class VoiceGender { FEMALE, MALE }

object GoogleTtsVoices {

    const val DEFAULT_VOICE_NAME = "de-DE-Chirp3-HD-Kore"
    const val DEFAULT_LANGUAGE_CODE = "de-DE"

    /** Alle 30 deutschen Chirp 3 HD-Stimmen (14 weiblich, 16 maennlich). */
    val ALL: List<GoogleTtsVoice> = listOf(
        // Weibliche Stimmen
        GoogleTtsVoice("de-DE-Chirp3-HD-Achernar", "Achernar — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Aoede", "Aoede — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Autonoe", "Autonoe — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Callirrhoe", "Callirrhoe — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Despina", "Despina — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Erinome", "Erinome — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Gacrux", "Gacrux — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Kore", "Kore — weiblich (Standard)", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Laomedeia", "Laomedeia — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Leda", "Leda — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Pulcherrima", "Pulcherrima — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Sulafat", "Sulafat — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Vindemiatrix", "Vindemiatrix — weiblich", VoiceGender.FEMALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Zephyr", "Zephyr — weiblich", VoiceGender.FEMALE),
        // Maennliche Stimmen
        GoogleTtsVoice("de-DE-Chirp3-HD-Achird", "Achird — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Algenib", "Algenib — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Algieba", "Algieba — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Alnilam", "Alnilam — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Charon", "Charon — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Enceladus", "Enceladus — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Fenrir", "Fenrir — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Iapetus", "Iapetus — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Orus", "Orus — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Puck", "Puck — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Rasalgethi", "Rasalgethi — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Sadachbia", "Sadachbia — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Sadaltager", "Sadaltager — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Schedar", "Schedar — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Umbriel", "Umbriel — männlich", VoiceGender.MALE),
        GoogleTtsVoice("de-DE-Chirp3-HD-Zubenelgenubi", "Zubenelgenubi — männlich", VoiceGender.MALE),
    )

    fun findByName(name: String): GoogleTtsVoice? = ALL.firstOrNull { it.name == name }

    fun displayNameFor(name: String): String =
        findByName(name)?.displayName ?: name
}

File: VoiceAgent/src/VoiceAgent/Services/GoogleTtsVoices.cs
using System.Collections.Generic;
using System.Linq;

namespace VoiceAgent.Services
{
    public enum VoiceGender { Female, Male }

    /// <summary>Eine auswaehlbare Google-TTS-Stimme.</summary>
    public sealed record GoogleTtsVoice(string Name, string DisplayName, VoiceGender Gender);

    /// <summary>
    /// Die 30 deutschen Google "Chirp 3 HD"-Stimmen (Franks "3D-Stimmen"), portiert aus
    /// dem EntropieReductor. Quelle: cloud.google.com/text-to-speech/docs/chirp3-hd.
    /// Default: de-DE-Chirp3-HD-Kore (weiblich, warm).
    /// </summary>
    public static class GoogleTtsVoices
    {
        public const string DefaultVoiceName = "de-DE-Chirp3-HD-Kore";
        public const string DefaultLanguageCode = "de-DE";

        public static readonly IReadOnlyList<GoogleTtsVoice> All = new List<GoogleTtsVoice>
        {
            // Weibliche Stimmen
            new("de-DE-Chirp3-HD-Achernar",     "Achernar — weiblich",        VoiceGender.Female),
            new("de-DE-Chirp3-HD-Aoede",        "Aoede — weiblich",           VoiceGender.Female),
            new("de-DE-Chirp3-HD-Autonoe",      "Autonoe — weiblich",         VoiceGender.Female),
            new("de-DE-Chirp3-HD-Callirrhoe",   "Callirrhoe — weiblich",      VoiceGender.Female),
            new("de-DE-Chirp3-HD-Despina",      "Despina — weiblich",         VoiceGender.Female),
            new("de-DE-Chirp3-HD-Erinome",      "Erinome — weiblich",         VoiceGender.Female),
            new("de-DE-Chirp3-HD-Gacrux",       "Gacrux — weiblich",          VoiceGender.Female),
            new("de-DE-Chirp3-HD-Kore",         "Kore — weiblich (Standard)", VoiceGender.Female),
            new("de-DE-Chirp3-HD-Laomedeia",    "Laomedeia — weiblich",       VoiceGender.Female),
            new("de-DE-Chirp3-HD-Leda",         "Leda — weiblich",            VoiceGender.Female),
            new("de-DE-Chirp3-HD-Pulcherrima",  "Pulcherrima — weiblich",     VoiceGender.Female),
            new("de-DE-Chirp3-HD-Sulafat",      "Sulafat — weiblich",         VoiceGender.Female),
            new("de-DE-Chirp3-HD-Vindemiatrix", "Vindemiatrix — weiblich",    VoiceGender.Female),
            new("de-DE-Chirp3-HD-Zephyr",       "Zephyr — weiblich",          VoiceGender.Female),
            // Maennliche Stimmen
            new("de-DE-Chirp3-HD-Achird",        "Achird — maennlich",        VoiceGender.Male),
            new("de-DE-Chirp3-HD-Algenib",       "Algenib — maennlich",       VoiceGender.Male),
            new("de-DE-Chirp3-HD-Algieba",       "Algieba — maennlich",       VoiceGender.Male),
            new("de-DE-Chirp3-HD-Alnilam",       "Alnilam — maennlich",       VoiceGender.Male),
            new("de-DE-Chirp3-HD-Charon",        "Charon — maennlich",        VoiceGender.Male),
            new("de-DE-Chirp3-HD-Enceladus",     "Enceladus — maennlich",     VoiceGender.Male),
            new("de-DE-Chirp3-HD-Fenrir",        "Fenrir — maennlich",        VoiceGender.Male),
            new("de-DE-Chirp3-HD-Iapetus",       "Iapetus — maennlich",       VoiceGender.Male),
            new("de-DE-Chirp3-HD-Orus",          "Orus — maennlich",          VoiceGender.Male),
            new("de-DE-Chirp3-HD-Puck",          "Puck — maennlich",          VoiceGender.Male),
            new("de-DE-Chirp3-HD-Rasalgethi",    "Rasalgethi — maennlich",    VoiceGender.Male),
            new("de-DE-Chirp3-HD-Sadachbia",     "Sadachbia — maennlich",     VoiceGender.Male),
            new("de-DE-Chirp3-HD-Sadaltager",    "Sadaltager — maennlich",    VoiceGender.Male),
            new("de-DE-Chirp3-HD-Schedar",       "Schedar — maennlich",       VoiceGender.Male),
            new("de-DE-Chirp3-HD-Umbriel",       "Umbriel — maennlich",       VoiceGender.Male),
            new("de-DE-Chirp3-HD-Zubenelgenubi", "Zubenelgenubi — maennlich", VoiceGender.Male),
        };

        public static GoogleTtsVoice? FindByName(string name) => All.FirstOrDefault(v => v.Name == name);

        public static string DisplayNameFor(string name) => FindByName(name)?.DisplayName ?? name;
    }
}
